package typed.entity;

// 人物信息
message Player
{
	required uint64	guid				= 1;
	optional string	name	 			= 2;
	optional uint32	level	 			= 3;
	optional uint32	exp	 				= 4;
	optional uint32	sex					= 5;
	optional uint32	profession			= 6;
	optional uint32	money	 			= 7;		// 铜钱
	optional uint32	coin	 			= 8;		// 元宝
	optional int32	src_x	 			= 9;		
	optional int32	src_y	 			= 10;
	optional int32	dst_x	 			= 11;
	optional int32	dst_y	 			= 12;
	optional uint32	seniority			= 13;		// 阅历
	optional uint32	vip_level			= 14;		
	optional uint32	coin_exp			= 15;		// 元宝消费积分
	optional uint32	prestige 			= 16;		// 声望
	optional uint32	energy	 			= 17;		// 体力
	optional uint32	power	 			= 18;		// 战力
	optional uint32	training_slot		= 19;		// 训练位置数量
	optional uint32	member_slot	 		= 20;		// 招募位置数量
	optional uint32	map_id	 			= 21;		// 地图id
	optional uint32 line_no				= 22;		// 所在线
	optional uint64	stage	 			= 23;		// 关卡guid
	optional int32	direction			= 24;		// 人物朝向
	optional uint64 form				= 25;		// 阵型
	optional uint32 rank				= 26;		// 排名
	optional uint64 guild				= 27;		// 公会ID
	optional string guild_name			= 28;		// 公会名称
	repeated uint32	wonder_skill		= 29;		// 玩家奇术等级
	repeated uint64	buffers			 	= 30;		// buffer 列表
	repeated uint64	roles	 			= 31;		// 角色列表
	repeated uint64	bag	 				= 32;		// 背包		0 : 锁定  1 ：解锁  其他：物品
	repeated uint64	storehouse			= 33;		// 仓库		0 : 锁定  1 ：解锁  其他：物品
	repeated uint32	hunt_master_state	= 34;		// 猎命师状态
	repeated uint32	can_recruit_roles	= 35;		// 道具解锁可招募角色
	repeated uint64	leave_team_roles	= 36;		// 已经离队角色
	optional uint64	sect	 			= 37;		// 门派
	optional uint64	law		 			= 38;		// 心法

	optional uint32 sthen_will_succ		= 39;		// 是否已经将强化成功率提升至100%
	repeated uint64 sthen_queue_stamp	= 40;		// 强化队列的冷却计时的起始时间点
	repeated uint32 sthen_queue_times	= 51;
	repeated uint32 sthen_queue_flags	= 52;
	optional uint64 stheno_hire_checksum	= 41;
	repeated uint32 stheno_hire_list	= 42;		// 待招募强化师列表，值为强化师id
	repeated uint64 stheno_idle_list	= 43;
	repeated uint64 stheno_equip_list	= 44;
	optional uint64 announce			= 45;		// 通告实体guid
	optional uint64 task_brief			= 46;		// 玩家task brief实体guid
	repeated uint32	attributes	 		= 47;
	optional uint32 level_limit			= 48;		// 等级上限
	repeated uint64 town_stage			= 49;		// 一个城镇一个town stage对象
	optional uint64 educate				= 50;		// 人物培养相关信息实体guid
	optional uint64 attach				= 53;
	optional uint64 specshop			= 54;		// 神秘商店
	optional uint64 chakra				= 55;		// 脉轮
	optional uint64 last_clean_day_relational_time	= 56;	// 最后一次清零天相关的资源
	optional uint64 last_login_time		= 57;		// 最后一次登陆时间	单位：秒
	optional uint64 last_logoff_time	= 58;		// 最后一次登出时间	单位：秒
	optional uint64 worship				= 59;		// worship实体guid
	optional uint64 role_id				= 60;
	optional uint64 first_role			= 61;
	optional uint64 last_clean_week_relational_time	= 62;	// 最后一次清零周相关的资源
	optional uint64 eggs				= 63;		// 武林大坏蛋
	optional uint64 condor				= 64;		// 射雕
	optional uint32 reverence			= 65;		// 威望
	optional uint32 vip_exp				= 66;		// vip exp
	optional uint64 boss				= 67;		// boos
	optional uint64 answer				= 68;		// 答题
	optional uint64 addiction			= 69;		// 防沉迷guid
	optional uint32 is_addiction		= 70;		// 是否防沉迷（0：不是，1：是）
	optional uint32 addiction_status	= 71;		// 防沉迷标志（0：正常，1：收益减半，2：收益为0）
	optional uint64 gift				= 72;		// 在线奖励实体
	optional uint64 vip					= 73;		// vip实体
	optional uint32 sthen_flags			= 74;
	optional string account				= 75;		// 账号
	optional uint32 super_gm			= 76;		// 是否超级GM账号, 1: yes
	optional uint64 achievement			= 77;		// 成就guid
	optional uint64 practice			= 78;		// 双修guid
	optional uint32 flower_rank			= 79;		// 0：无排名，1、2、3排名
	optional uint64 daily_points		= 80;		// 每日奖励guid
	optional uint64 tower				= 81;		// 推塔
	optional uint32 start_stheno		= 82;		// 记录初始化强化师
	optional uint64 tavern				= 83;		// 酒馆guid
	optional uint64 fish				= 84;		// 钓鱼guid
	optional uint64 dodge				= 85;		// 轻功guid	
	optional uint32 guild_icon			= 86;		// 工会图标ID
	repeated uint64	wifes	 			= 87;		// 角色列表
	optional uint64 player_wife			= 88;		// 侠侣
	optional uint64 player_activies		= 89;		// 活动模板
	optional uint64 last_energy_reward_time	= 90;	// 最后一次获取精力充沛时间
	optional uint32 left_energy_reward		 = 91;	// 当前剩余精力充沛体力
	optional uint64 maze				= 92;		// 迷宫实体
	optional uint64 open_activity		= 93;		// 开服活动，充钱
	optional uint64 open_act			= 94;		// 开服活动，参与中奖
	optional uint64 coin_guid			= 95;		// coin guid
	optional string last_logoff_time_str	= 96;		// 最后一次登出时间，字符串类型
	repeated uint32 give_equip			= 97;		// 1 装备可取，2 装备已取
	repeated uint32 equip_reward		= 98;		// 1 奖励可取，2 奖励已取
	optional uint32 follow_role_id		= 99;		// 跟随伙伴ID
	optional uint32 max_power			= 100;		// 最高战力
	repeated uint32 item_id				= 101;		// 限购物品id((每日))
	repeated uint32 item_num			= 102;		// 限购物品已购买数量((每日))
	repeated uint32 current_title		= 103;		// 当前已经装备的称号
	optional uint32 enable_title		= 104;		// 当前属性生效的称号
	repeated uint32 push_id_list		= 105;		// 推送系统id
	repeated uint32 push_status_list	= 106;		// 推送系统状态 1：解锁未购买 2：解锁已购买
	optional uint32 jifen				= 107;		// 玩家积分,货币一种	
	
	optional uint64 total_money_cost	= 108;
	optional uint64 total_coin_cost		= 109;
	optional uint64 total_seni_cost		= 110;
	optional uint32 charge_quest		= 111;		// 玩家充值任务,1：未解锁，2：已解锁；3：已完成，4：已领取
	optional uint64 trial				= 112;		// 试炼
	optional uint64 runes				= 113;		// 符文
	optional uint64 shl_switch			= 114;		// 圣火令 开关
	optional uint64 doubleexp			= 115;		// 双倍经验
	optional uint64 soul_pve_guid		= 116;		// 武魂pveguid
	optional uint64 soul				= 117;		// 武魂系统
	optional uint32 soul_frag			= 118;		// 武魂碎片 数量
	optional uint64 login_gift			= 119;		// 三日登录领奖
	optional uint64 super_power			= 120;		// 超级战力(target系统)
	optional string platform			= 121;		// 平台名
	optional uint64 goldcard			= 122;		// 特权卡guid
	optional uint32 special_right		= 123;		// 特权卡通告
	optional uint64 talent				= 124;		// 伙伴资质
	optional uint32 turntable_accrun	= 125;		// 连抽次数
	repeated uint64 touch_times			= 126;		// 触发推送的时间
	optional string validate_str		= 127;		// 初始vip验证字符串
	optional uint32 vip_level_init		= 128;		// 初始vip等级
	optional uint32 year_card			= 129;		// 初始平台年卡
	optional uint32 time_stamp			= 130;		// 初始加密时间戳
	optional uint64 horse				= 131;		// 坐骑
	optional string robot_validate		= 132;		// 机器人验证码
	optional uint32	robot_id			= 133;		// 初始机器人id，0表示正常用户
	optional uint32 horse_id			= 134;		// 正在骑乘的马id
	optional uint32 wife_follow			= 135;		// 跟随的侠侣
	optional uint32 is_yellow_vip		= 136;		// 是否为黄钻用户, 0 : 不是, 1:是
	optional uint32 is_yelow_year_vip	= 137;		// 是否为年费黄钻用户, 0 : 不是, 1:是
	optional uint32 yellow_vip_level	= 138;		// 黄钻等级
	optional uint32 is_yellow_high_vip	= 139;		// 是否为豪华版黄钻用户 0: 不是, 1:是
	optional string open_id_qq			= 140;		// qq开放平台open id
	optional string open_id_referrer	= 141;		// qq好友邀请介绍人open id，没有填空
	optional uint64 qqspreadsen			= 142;		// qq 推广活动sen实体
	optional uint32 wf_fatepoint		= 143;		// 侠侣缘分点
}

// 角色信息
message Role
{
	required uint64	guid	 			= 1; 
	required uint64	player	 			= 2; 
	optional uint32	role_id		 		= 3;
	optional string	name	 			= 4;
	optional uint32	sex	 				= 5;
	optional uint32	job	 				= 6;		// 角色职业
	optional uint32	faction	 			= 7;		// 门派
	optional uint32	level	 			= 8;
	optional uint32	exp	 				= 9;
	optional uint32	postion				= 10;		// 当前阵型中的位置
	optional uint32	normal_skill 		= 11;		// 普通技能ID
	optional uint32	extremely_skill		= 12;		// 绝技技能ID
	optional uint32 char_ability		= 13;		// 被动技能
	repeated uint32	culture	 			= 14;		// 培养
	repeated uint32	pellet1	 			= 15;		// 丹药，不使用
	repeated uint32	pellet_no	 		= 16;		// 丹药吃掉的个数，不使用
	repeated int32	attributes	 		= 17;
	repeated uint64	buffers	 			= 18;
	repeated uint64	equipments	 		= 19;		// 装备位
	repeated uint64	hunt	 			= 20;		// 猎命
	optional uint32	hire_id		 		= 21;
	optional uint32 avatar_id			= 22;		// avatar id
	optional uint64 educate				= 23;		// 角色培养实体guid
	optional uint32 stomach_start_time	= 24;		// 大胃值开始时间，为0则表示大胃值满
	optional uint32 quality				= 25;		// 伙伴品质
	optional uint32 training_level		= 26;		// 伙伴强化等级
	optional uint32 dodge_skill_id		= 27;		// 轻功主动技能
	optional uint64 wife				= 28;		// 关联的侠侣。0表示没有。
	optional uint64 pellet				= 29;		// 丹药
	optional uint32 inherit_to			= 30;		// 传承给其他伙伴次数
	optional uint32 inherit_from		= 31;		// 从其他伙伴传承过来次数
	optional uint64	shl_owner			= 32;		// 圣火令物主
	optional uint64 update_fight_power	= 33;
}

message Coin
{
	required uint64 guid				= 1;
	required uint64 player				= 2;
	optional uint32 coin				= 3;
	optional uint32 money				= 4;
}